### PR TITLE
Enhance error report of Bootstrap demo server.

### PR DESCRIPTION
**Pull request by sbernard31**

### CODE REVIEW

File: leshan-bsserver-demo/src/main/java/org/eclipse/leshan/server/bootstrap/demo/servlet/BootstrapServlet.java [hunk 1]
```diff
@@ -140,7 +140,13 @@ protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws
...
             resp.setStatus(HttpServletResponse.SC_NO_CONTENT);
         } else {
-            resp.sendError(HttpServletResponse.SC_NOT_FOUND);
+            sendError(resp, HttpServletResponse.SC_NOT_FOUND, "no config for " + endpoint);
         }
     }
+
+    private void sendError(HttpServletResponse resp, int statusCode, String errorMessage) throws IOException {
+        resp.setStatus(statusCode);
+        if (errorMessage != null)
```
Comment (scop) [inline leshan-bsserver-demo/src/main/java/org/eclipse/leshan/server/bootstrap/demo/servlet/BootstrapServlet.java]: I suggest placing `resp.setContentType("text/plain; charset=UTF-8")` here.

File: leshan-bsserver-demo/src/main/java/org/eclipse/leshan/server/bootstrap/demo/servlet/BootstrapServlet.java [hunk 2]
```diff
@@ -140,7 +140,13 @@ protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws
...
+            resp.getOutputStream().write(errorMessage.getBytes(StandardCharsets.UTF_8));
```
Comment (scop) [inline leshan-bsserver-demo/src/main/java/org/eclipse/leshan/server/bootstrap/demo/servlet/BootstrapServlet.java]: Maybe `flush()` the output stream after this?

File: leshan-bsserver-demo/src/main/java/org/eclipse/leshan/server/bootstrap/demo/servlet/BootstrapServlet.java (hunk 1 above)
Comment (sbernard31) [inline leshan-bsserver-demo/src/main/java/org/eclipse/leshan/server/bootstrap/demo/servlet/BootstrapServlet.java]: I added the content type.

File: leshan-bsserver-demo/src/main/java/org/eclipse/leshan/server/bootstrap/demo/servlet/BootstrapServlet.java (hunk 2 above)
Comment (sbernard31) [inline leshan-bsserver-demo/src/main/java/org/eclipse/leshan/server/bootstrap/demo/servlet/BootstrapServlet.java]: I don't think we need to flush the stream : https://stackoverflow.com/questions/5043657/do-i-need-to-flush-the-servlet-outputstream

**Comment by sbernard31** [comment]
Thx @scop, integrated in master (commit f95d1da478ab121ec9145e3ae44a596d05c0a211)

**Pull request closed without merging**